ports/mesa/portfile.cmake

```
@@ -0,0 +1,169 @@
+            HEAD_REF master
+        )
+        execute_process(COMMAND ${PYTHON3_DIR}/python${VCPKG_HOST_EXECUTABLE_SUFFIX} ${PYFILE_PATH}/get-pip.py ${PYTHON_OPTION})
+    endif()
+    execute_process(COMMAND ${PYTHON3_DIR}/Scripts/pip${VCPKG_HOST_EXECUTABLE_SUFFIX} install mako ${PYTHON_OPTION})
+    execute_process(COMMAND ${PYTHON3_DIR}/Scripts/pip${VCPKG_HOST_EXECUTABLE_SUFFIX} install setuptools ${PYTHON_OPTION})
+else()
+    execute_process(COMMAND ${PYTHON3_DIR}/easy_install${VCPKG_HOST_EXECUTABLE_SUFFIX} mako)
+    execute_process(COMMAND ${PYTHON3_DIR}/easy_install${VCPKG_HOST_EXECUTABLE_SUFFIX} setuptools)
+endif()
+if(NOT VCPKG_TARGET_IS_WINDOWS)
+    execute_process(COMMAND pip3 install setuptools mako)
+endif()
+
+vcpkg_find_acquire_program(FLEX)
+get_filename_component(FLEX_DIR "${FLEX}" DIRECTORY )
+vcpkg_add_to_path(PREPEND "${FLEX_DIR}")
+vcpkg_find_acquire_program(BISON)
+get_filename_component(BISON_DIR "${BISON}" DIRECTORY )
+vcpkg_add_to_path(PREPEND "${BISON_DIR}")
+
+if(WIN32) # WIN32 HOST probably has win_flex and win_bison!
+    if(NOT EXISTS "${FLEX_DIR}/flex${VCPKG_HOST_EXECUTABLE_SUFFIX}")
+        file(CREATE_LINK "${FLEX}" "${FLEX_DIR}/flex${VCPKG_HOST_EXECUTABLE_SUFFIX}")
+    endif()
+    if(NOT EXISTS "${BISON_DIR}/BISON${VCPKG_HOST_EXECUTABLE_SUFFIX}")
+        file(CREATE_LINK "${BISON}" "${BISON_DIR}/bison${VCPKG_HOST_EXECUTABLE_SUFFIX}")
+    endif()
+endif()
+
+# For features https://github.com/pal1000/mesa-dist-win should be probably studied a bit more. 
+#string(APPEND GALLIUM_DRIVERS 'auto')
+list(APPEND MESA_OPTIONS -Dopengl=true)
+list(APPEND MESA_OPTIONS -Dzstd=enabled)
+list(APPEND MESA_OPTIONS -Dshared-llvm=auto)
+list(APPEND MESA_OPTIONS -Dlibunwind=disabled)
+list(APPEND MESA_OPTIONS -Dlmsensors=disabled)
+list(APPEND MESA_OPTIONS -Dvalgrind=disabled)
+list(APPEND MESA_OPTIONS -Dglvnd=false)
+list(APPEND MESA_OPTIONS -Dglx=disabled)
+list(APPEND MESA_OPTIONS -Dgbm=disabled)
+#list(APPEND MESA_OPTIONS -Dosmesa=['gallium','classic']) # classic has compiler errors. 
+list(APPEND MESA_OPTIONS -Dosmesa=['gallium'])
+
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    list(APPEND MESA_OPTIONS -Dshared-swr=false)
+    list(APPEND MESA_OPTIONS "-Dswr-arches=['avx']")
+else()
+    list(APPEND MESA_OPTIONS -Dshared-swr=true)
+    list(APPEND MESA_OPTIONS "-Dswr-arches=['avx','avx2','knl','skx']")
+endif()
+
+string(APPEND GALLIUM_DRIVERS 'swrast')
+if("llvm" IN_LIST FEATURES)
+    list(APPEND MESA_OPTIONS -Dllvm=enabled)
+    string(APPEND GALLIUM_DRIVERS ",'swr'") # SWR always requires llvm
+else()
+    list(APPEND MESA_OPTIONS -Dllvm=disabled)
+endif()
+if("shared-glapi" IN_LIST FEATURES)
+    list(APPEND MESA_OPTIONS -Dshared-glapi=enabled)  #shared GLAPI required when building two or more of the following APIs - opengl, gles1 gles2
+else()
+    list(APPEND MESA_OPTIONS -Dshared-glapi=disabled)
+endif()

```

```suggestion
foreach(feature IN LISTS FEATURES)
    if(feature IN_LIST "opengl;gles1;gles2")
        list(APPEND GL_FEATURES "${feature}")
    endif()
    list(LENGTH GL_FEATURES NUMBER_GL_FEATURES)
endforeach()

if(NUMBER_GL_FEATURES GREATER 1)
    list(APPEND MESA_OPTIONS -Dshared-glapi=enabled)
else()
    list(APPEND MESA_OPTIONS -Dshared-glapi=disabled)
endif()
```

something like this?

---

I kept sharedgl-api as a feature but made it fatally error if more than one api is selected and it is disabled.

---

it is the default now.